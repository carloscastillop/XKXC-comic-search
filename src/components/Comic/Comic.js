import React from 'react';
import style from './Comic.module.scss';
import Image from './Image/Image';
import PropTypes from 'prop-types';
import Pagination from './Pagination/Pagination';
import {connect} from 'react-redux';

const Comic = ({comic: {title, alt, img, num}, latestComic}) => {
    const caption = (
        <>
            <title>{title}</title>
            <span>
                {alt}
            </span>
        </>
    );
    const last = (latestComic)? latestComic.num : 0;
    const pagination = (num) ? <Pagination num={num} latest={last}/> : null;

    return (
        <section className={style.comicContainer} data-testid={'comic'}>
            <div className={style.container}>
                <h1 className={`${style.h2} ${style['my-4']} animated fadeInUp`}>
                    {title}
                </h1>
                <h4 className={'animated fadeIn'}>
                    <span className={`${style['badge']} ${style['badge-secondary']}`}>
                        #{num}
                    </span>
                </h4>
                <div className={style['mb-3']}>
                    <a
                        href={`//xkcd.com/${num}`}
                        title={title}
                        rel="noreferrer"
                    >
                        <Image
                            imageSrc={img}
                            imageTitle={title}
                            imageAlt={title}
                            imageCaption={caption}
                        />
                    </a>
                </div>
                <div>
                    <p className={'animated fadeInUp'}>
                        {alt}
                    </p>
                </div>
            </div>
            {pagination}
        </section>
    );
};

Comic.propTypes = {
    comic: PropTypes.object.isRequired,
    latestComic: PropTypes.object.isRequired
};

const mapStateToProps = state => {
    return {
        latestComic: state.latestComic
    };
};

export default connect(
    mapStateToProps
)(Comic);