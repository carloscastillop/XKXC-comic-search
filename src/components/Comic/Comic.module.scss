@import '../../styles/global.module';

.comicContainer {
  text-align: center;
  h1 {
    color: #fff;
  }
  p {
    color: rgba(255,255,255,0.5);
  }
}