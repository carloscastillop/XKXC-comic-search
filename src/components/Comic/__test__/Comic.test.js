import React from 'react';
import {render} from '@testing-library/react';
import Comic from '../Comic';

test('Get Comic container', () => {
    const {getByTestId} = render(<Comic/>);
    const linkElement = getByTestId('comic');
    expect(linkElement).toBeInTheDocument();
});
