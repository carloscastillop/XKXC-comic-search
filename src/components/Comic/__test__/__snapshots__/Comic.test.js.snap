// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comic snapshop 1`] = `
<section
  className="comicContainer"
  data-testid="comic"
>
  <div
    className="container"
  >
    <h1
      className="h2 my-4 animated fadeInUp"
    />
    <h4
      className="animated fadeIn"
    >
      <span
        className="badge badge-secondary"
      >
        #
      </span>
    </h4>
    <div
      className="mb-3"
    >
      <a
        href="//xkcd.com/undefined"
        rel="noreferrer"
      >
        <div
          className="imageContainer"
          data-testid="image"
        >
          <span
            className=" lazy-load-image-background blur"
            style={
              Object {
                "backgroundImage": "",
                "backgroundSize": "",
                "color": "transparent",
                "display": "inline-block",
                "height": undefined,
                "width": undefined,
              }
            }
          >
            <span
              className="img-fluid"
              style={
                Object {
                  "display": "inline-block",
                }
              }
            />
          </span>
          <div
            className="imageCaptionContainer"
          >
            <div
              className="imageCaption"
            >
              <title />
              <span />
            </div>
          </div>
        </div>
      </a>
    </div>
    <div>
      <p
        className="animated fadeInUp"
      />
    </div>
  </div>
</section>
`;
